plugins/schedule/base/lib/kScheduleEventsConsumer.php

```diff
@@ -120,19 +172,74 @@ public function updateScheduleEvent($eventId)
-            foreach ($scheduleEvents as $currScheduleEvent)
-            {
-                /**
-                 * @var ScheduleEventResource $currScheduleEvent
-                 */
-                $currScheduleEvent->delete();
-            }
-        }
-    }
+	protected function scheduleEventChanged(ScheduleEvent $scheduleEvent, $modifiedColumns = null)
+	{
+		if (in_array($scheduleEvent->getStatus(), array(ScheduleEventStatus::DELETED, ScheduleEventStatus::CANCELLED)))
+		{
+			$this->scheduleEventChangedEventDeletedOrCanceled($scheduleEvent, $modifiedColumns);
+		}
+		if ($modifiedColumns)
+		{
+			if (in_array(ScheduleEventPeer::END_DATE, $modifiedColumns))
+			{
+				$this->scheduleEventChangedEndDateChanged($scheduleEvent, $modifiedColumns);
+			}
+			if (in_array(ScheduleEventPeer::CUSTOM_DATA, $modifiedColumns))
+			{
+				$this->scheduleEventChangedLinkedToChanged($scheduleEvent, $modifiedColumns);
+			}
+		}
+	}
+	
+	protected function scheduleEventChangedEventDeletedOrCanceled(ScheduleEvent $scheduleEvent, $modifiedColumns = null)
+	{
+		$scheduleEvents = ScheduleEventResourcePeer::retrieveByEventId($scheduleEvent->getId());
+		foreach ($scheduleEvents as $currScheduleEvent)
+		{
+			/**
+			 * @var ScheduleEventResource $currScheduleEvent
+			 */
+			$currScheduleEvent->delete();
+		}
+		if ($scheduleEvent->getLinkedTo() && !is_null($scheduleEvent->getLinkedTo()->getEventId()))
+		{
+			$scheduleEvent->removeCurrentEventFromPrecedingEvent($scheduleEvent->getLinkedTo()->getEventId());
+		}
+		if ($scheduleEvent->getLinkedBy())
+		{
+			$scheduleEvent->unlinkFollowerEvents();
+		}
+	}
+	
+	protected function scheduleEventChangedEndDateChanged(ScheduleEvent $scheduleEvent, $modifiedColumns = null)
+	{
+		//update start & end date for all linked by events
+		$scheduleEvent->updateStartEndTimeOfFollowerEvents();
+		if ($scheduleEvent->getLinkedTo() && $scheduleEvent->getLinkedTo()->getEventId())
+		{
+			$linkedToStartDate = $scheduleEvent->getLinkedToEndTime();
+			$offset = $scheduleEvent->getLinkedTo()->getOffset();
+			if (strtotime($scheduleEvent->getStartDate()) != strtotime($linkedToStartDate) + $offset)
+			{
+				$scheduleEvent->removeCurrentEventFromPrecedingEvent();
+				$scheduleEvent->setLinkedTo(null);
+				$scheduleEvent->save();
+			}
+		}
+	}
+	
+	protected function scheduleEventChangedLinkedToChanged(ScheduleEvent $scheduleEvent, $modifiedColumns = null)
+	{
+		$oldCustomData = $scheduleEvent->getCustomDataOldValues();
+		if ($scheduleEvent->getLinkedTo() && is_null($scheduleEvent->getLinkedTo()->getEventId()) &&
+			(array_key_exists('linkedTo', $oldCustomData['']) && !is_null($oldCustomData['']['linkedTo']->getEventId())))
+		{
+			$scheduleEvent->removeCurrentEventFromPrecedingEvent($oldCustomData['']['linkedTo']->getEventId());
+		}
+		if ($scheduleEvent->getLinkedTo() && $scheduleEvent->getLinkedTo()->getEventId())

```

Comment:
what if the value is here - but it was never changed?

Reply:
From what i saw, the $scheduleEvent object contains the new values that were changed. 
In any case, if the id is already in the  linkedBy field, it wont be added again.